pil_analyzer/src/evaluator.rs

```diff
@@ -248,13 +248,17 @@ mod internal {
                     .collect::<Result<_, _>>()?,
             ),
             Expression::BinaryOperation(left, op, right) => {
-                let left = evaluate(left, locals, symbols)?;
-                let right = evaluate(right, locals, symbols)?;
-                match (&left, &right) {
-                    (Value::Custom(_), _) | (_, Value::Custom(_)) => {
+                let mut left = evaluate(left, locals, symbols)?;
+                let mut right = evaluate(right, locals, symbols)?;
+                match (&mut left, op, &mut right) {
+                    (Value::Custom(_), _, _) | (_, _, Value::Custom(_)) => {
                         symbols.eval_binary_operation(left, *op, right)?
                     }
-                    (Value::Number(l), Value::Number(r)) => {
+                    (Value::Array(l), BinaryOperator::Add, Value::Array(r)) => {
```

Comment:
Any reason not to allow `+` to implement concatenation on arrays?

Reply:
no